Add WindowsFirewall workbook

TheAwesem opened the pull request:
Workbook to visualise data from the WindowsFirewall in correlation with security and azure signin logs

Submission for reward #522

shainw commented on Workbooks/WindowsFirewall.json:
```diff
@@ -0,0 +1,524 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "let param_Computers = \"{Computers}\";\r\nWindowsFirewall\r\n| where (Computer == param_Computers or param_Computers contains Computer or param_Computers == \"\") and SourceIP !in (\"::1\",\"-\")\r\n| summarize FirewallEvents = count() by SourceIP\r\n| join(\r\nSecurityEvent\r\n| where IpAddress != \"\" and (Computer == param_Computers or param_Computers contains Computer or par ...
```
- Use isempty() versus == "" - for example - isempty(param_Computers)
- Use isnotempty() versus != "" - for example - isnotempty(IpAddress)

TheAwesem commented:
@shainw Thank you for the detailed feedback.
I have implemented the changes and fixed some other issues with the parameters and scaling.
Also I added the updated preview images. 
I hope the issues are fixed and I look forward to your feedback

shainw commented on Workbooks/WindowsFirewall.json:
```diff
@@ -0,0 +1,541 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "let param_Computers = \"{Computers}\";\r\nWindowsFirewall \r\n| where (Computer == param_Computers or param_Computers contains Computer or isempty(param_Computers))\r\n| summarize FirewallEvents = count() by SourceIP\r\n| join kind = inner(\r\n   SigninLogs\r\n   | summarize SuccessAzureLogin = countif(ResultType == 0), FailedAzureLogin = countif(ResultType != 0) by SourceIP = I ...
```
Missed adding the filter for LogonType == 10 for Security Events so that you only get RDP logon attempts.

TheAwesem replied:
I forgot to add to this query. Only applied it to the other query.

juliango2100 commented on Workbooks/WindowsFirewall.json:
```diff
@@ -0,0 +1,541 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "let commonPorts = dynamic({\"443\": \"HTTPS\", \"80\":\"HTTP\", \"3389\":\"RDP\", \"53\":\"DNS\", \"389\":\"LDAP\", \"445\":\"SMB\", \"135\":\"RPC\", \"47001\":\"WinRM\",\"22\":\"ssh\", \"21\": \"ftp\"}); // Set of common portnames\r\nlet param_Computers = \"{Computers}\";\r\nWindowsFirewall\r\n| where (Computer == param_Computers or param_Computers contains Computer or isempty( ...
```
ICMP will show up in the firewall logs and will not have a destination port.  I suggest filtering out ports that are empty.  Example:
WindowsFirewall
| extend destinationport = tostring(DestinationPort)
| where strlen(dp) > 0

TheAwesem replied:
Used isnotemtpy(DestinationPort) to keep the same syntax as the other changes. Also applied to the Piechart by protocol

juliango2100 commented on Workbooks/WindowsFirewall.json:
```diff
@@ -0,0 +1,541 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "let param_Computers = \"{Computers}\";\r\nSecurityEvent\r\n| where AccountType == \"User\" and  isnotempty(IpAddress) and (Computer == param_Computers or param_Computers contains Computer or param_Computers == \"\")\r\n| summarize EventCount = count(), DistinctIPCount = dcount(IpAddress),IPAddresses = makeset(IpAddress) by Account\r\n| top 10 by DistinctIPCount desc\r\n| project ...
```
If 2 or more machines have the same local username, the results will be merged which is incorrect.
Suggestion: Append the machine name to the user name so that distinct user names are shown in the table.

TheAwesem replied:
Added Computer to the summarize and project.
I choose to add Computer to project instead of a strcat() to keep the information 
better readable

TheAwesem commented:
@juliango2100 Thank you for your feedback, I have modified the query's with some minor changes to the suggestions. I hope these changes solve the problems.
